src/main/java/org/elasticsearch/index/query/ExistsQueryParser.java

```diff
@@ -19,21 +19,15 @@
 
 package org.elasticsearch.index.query;
 
-import org.apache.lucene.search.*;
 import org.elasticsearch.common.inject.Inject;
-import org.elasticsearch.common.lucene.search.Queries;
 import org.elasticsearch.common.xcontent.XContentParser;
-import org.elasticsearch.index.mapper.FieldMapper;
-import org.elasticsearch.index.mapper.MapperService;
-import org.elasticsearch.index.mapper.internal.FieldNamesFieldMapper;
 
 import java.io.IOException;
-import java.util.Collection;
 
 /**
  *
  */
-public class ExistsQueryParser extends BaseQueryParserTemp {
+public class ExistsQueryParser extends BaseQueryParser {
 
     @Inject
```

Comment:
I think so, at least we have it on a couple of other no-arg parsers. I'll try what happens when I remove them but I guess it messes up some guice registry stuff.